src/sentry/integrations/slack/requests/base.py

```diff
@@ -120,17 +123,22 @@ def logging_data(self) -> Mapping[str, str]:
 
         return {k: v for k, v in data.items() if v}
 
-    def get_identity(self) -> Identity | None:
-        try:
-            idp = IdentityProvider.objects.get(type="slack", external_id=self.team_id)
-        except IdentityProvider.DoesNotExist as e:
-            logger.info("slack.action.invalid-team-id", extra={"slack_team": self.team_id})
-            raise e
-
-        try:
-            return Identity.objects.select_related("user").get(idp=idp, external_id=self.user_id)
-        except Identity.DoesNotExist:
+    def get_identity(self) -> APIIdentity | None:
+        provider = identity_service.get_provider(
+            provider_type="slack", provider_ext_id=self.team_id
+        )
+        self._identity = (
```

Comment:
This is a minor nit, but is it possible to wrap this in a `if not hasattr(self, '_identity')`?  Making it fully idempotent.

As written, if the identity is `None` on one `get_identity` call, the next `get_identity` call can still make another remote call, over and over.  This can be a performance issue on multiple accesses.

This would also simplify the code below

Reply:
great catch!